src/pages/Location/Location.js

```diff
@@ -260,6 +344,129 @@ const LocationComponent = () => {
+          if (!field.id) {
+            return postFields(accessTokenRedux, field);
+          }
+          return null;
+        });
+        await Promise.all(fieldUpdates);
+        getFields(accessTokenRedux).then((data) => {
+          // TODO: since all updated fields are send to the backend after leave the page,
+          // the backend time might be not accurate to when show the field is updated
+          // data = {
+          //   ...data,
+          //   data: data.data.sort((a, b) => new Date(a.updatedAt) - new Date(b.updatedAt)),
+          // };
+          // console.log('sortedData', data);
+          dispatchRedux(updateField(data));
+        });
+      };
+      updateFields();
+    }
+  }, []);
+
+  // console.log('currArea', currArea, currArea?.features?.slice(-1)[0]);
+  // console.log('features', features);
+
+  const onDraw = (draw) => {
+    // console.log(draw.mode, draw.e.type, draw);
+    // set isaddingPoint to false
+    if (isAuthenticated && draw.mode !== 'select') {
+      setIsAddingPoint(false);
+      setFieldDialogState({
+        ...fieldDialogState, open: true, actionType: draw.mode, areaType: 'Polygon',
+      });
+    }
+  };
+
+  const handleClose = (save) => {
+    const { actionType, areaType, fieldName } = fieldDialogState;
+    if (save) {
+      if (actionType === 'add') {
+        const errText = fieldNameValidation(fieldName, userFieldsRef.current);
+        if (errText !== '') {
+          setFieldDialogState({ ...fieldDialogState, error: true, errorText: errText });
+          return;
+        }
+        const { longitude, latitude } = selectedToEditSite;
+        const point = buildPoint(longitude, latitude, fieldName);
+        // console.log('add area', polygon);
+        if (areaType === 'Point') {
+          userFieldsRef.current = [...userFieldsRef.current, point];
+        }
+        if (areaType === 'Polygon') {
+          const polygon = currArea.features.slice(-1)[0];
+          const geoCollection = buildGeometryCollection(point.geometry, polygon.geometry, fieldName);
+          userFieldsRef.current = [...userFieldsRef.current, geoCollection];
+        }
+        setSelectedUserField(fieldName);
+      }
+      if (actionType === 'update') {
+        // Only supports polygon updates
+        const { longitude, latitude } = selectedToEditSite;
+        const point = buildPoint(longitude, latitude, selectedUserField);
+        const polygon = currArea.features.slice(-1)[0];
+        // console.log('update', polygon);
+        const geoCollection = buildGeometryCollection(point.geometry, polygon.geometry, selectedUserField);
+        userFieldsRef.current = userFieldsRef.current.map((f) => {
+          if (f.label === selectedUserField) return geoCollection;
+          return f;
+        });
+      }
+      if (actionType === 'delete') {
+        // add a delete flag so when saving the code knows delete which field
+        userFieldsRef.current = userFieldsRef.current.map((f) => {
```

Comment:
all of these maps should have the "f" prop from the map changed to something more readable.  i suggest "userFieldRef" or "userField" ill leave comments on the others i found too.